rfcs/0134-nix-store-layer.md

```diff
@@ -0,0 +1,271 @@
+---
+feature: nix-store-layer
+start-date: 2022-09-06
+author: John Ericson
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Allow building a Nix executable with just the store layer.
+Do this to manage complexity, and promote a "Greater Nix" community.
+
+# Motivation
+[motivation]: #motivation
+
+## Managing complexity
+
+We have recently added a ton of functionality to Nix.
+Most of this is in Flakes, which just has enormous surface area, but things like floating content-addressed derivations and other RFCs add complexity to the core store layer too.
+
+If we view Nix as one monolithic whole, it will grow too complex and unwieldy, and we will be unable to manage it as we the complexity bogs us down.
+However, if we embrace layering we can "divide and conquer" the project, and manage that complexity.
+This will ensure the continued sustainability of Nix.
+
+We currently embrace layering somewhat as an implementation detail, but only as an implementation detail.
+The division between `libnixstore`, `libfetchers`, `libexpr`, etc. is not yet exposed to users, emphasized in documentation (though this is changing thanks to @Fricklerhandwerk's efforts with new [architecture documentation](https://github.com/NixOS/nix/pull/6888)!).
+
+We should instead fully embrace it:
+
+- Docs
+
+  - More advanced documentation can explain layering for those that want a deeper understanding of Nix.
+
+  - Even more basic documentation can still benefit from separate terminology before the layering is fully explained.
+    See https://www.haskellforall.com/2022/08/stop-calling-everything-nix.html for a phenomenal take-down of how calling everything "Nix" today confuses users and leaves them unable to articulate what parts of the ecosystem are frustrating then.
+
+- Separate executable ensure lower layers build in isolation, new integration tests those executable without aid of high-layer info.
```

Comment:
I'm having difficulty parsing this, maybe this sentence could do with a rewrite.

Reply:
How's this?
```suggestion
- Separate executables to ensure lower layers build in isolation without higher layers,

   - Integration tests for those executables that also don't use the higher layers to ensure those executables work in isolation. 
```
Frankly this is somewhat a restatement of the detailed design so I would be fine cutting it too,